src/lib/dialog/dialog.ts

```diff
@@ -91,11 +91,19 @@ export class MdDialog {
    */
   open<T>(componentOrTemplateRef: ComponentType<T> | TemplateRef<T>,
           config?: MdDialogConfig): MdDialogRef<T> {
+
+    const inProgressDialog = this._openDialogs.find(dialog => dialog._isAnimating());
```

Comment:
Do you think it's worthwhile to key this on the component as well? E.g., if I try to `open(UserProfile)` multiple times in a row, they get blocked, but if I try to `open(UserProfile)` and `open(AdminSettings)` at the same time then they both open?